pkg/dhcp6client/packetSock.go

```diff
@@ -54,6 +55,12 @@ func (pc packetSock) Close() error {
 	return syscall.Close(pc.fd)
 }
 
+// Set a read timeout
```

Comment:
this is nice, it's using kernel mechanisms. My one question here: since this is a packet socket, the same timeout idea should work for v4 and v6 sockets? If so, is there a way to write this so we have one function to set timeouts that we can use on any packet socket?